### PR TITLE
Added proofreader validation

### PR DESCRIPTION
**Pull request by seanwiseman**
I have removed explicit calls to `pylint` and `flake8` in `project_tests.sh` in favor of using [proofreader](https://github.com/elifesciences/proofreader-python). 

I have done some minor refactoring to align the project with the stricter defaults provided by `proofreader`. Therefore I have removed the local `.pylintrc` and `.flake8` override files as they are no longer needed.

### CODE REVIEW

**Comment by giorgiosironi** [comment]
There are some conflicts to solve, needs a merge of `master` into this branch.

**Review by giorgiosironi — APPROVED** [review]

File: elife_bus_sdk/__init__.py [hunk 1]
```diff
@@ -13,18 +13,18 @@
...
     Publisher factory function.
 
     If a publisher already exists with the `pub_name` value
     then you will be returned that instance, otherwise the publisher
     will be created and returned.
 
+    :param config: dict
     :param pub_name: str
     :param pub_type: str
-    :param config: dict
```
Comment (thewilkybarkid) [inline elife_bus_sdk/__init__.py:27]: These are all redundant anyway?

**Review by thewilkybarkid — APPROVED** [review]

File: elife_bus_sdk/__init__.py (hunk 1 above)
Comment (seanwiseman) [reply elife_bus_sdk/__init__.py:27]: On the surface, it looks like they duplicate type hinting, but they allow the generation of documentation which you could then [host](https://readthedocs.org/). Even if it's not going to be used immediately it's nice to have, it's a habit but has a use.

Comment (thewilkybarkid) [reply elife_bus_sdk/__init__.py:27]: Hmm, sounds like a tooling issue.

**Pull request merged**